lecture1/task1.c

```diff
@@ -0,0 +1,120 @@
+#include <stdio.h>
+#include <stdint.h>
+
+uint16_t swap16(uint16_t a)
+{
+	uint16_t res;
+	res = (a&0x00FF) << 8;
+	res += (a&0xFF00) >> 8;
+	return res;
+}
+uint32_t swap32(uint32_t a)
+{
+	uint32_t res;
+	res = (a&0x000000FF) << 24;
+	res += (a&0x0000FF00) << 8;
+	res += (a&0x00FF0000) >> 8;
+	res += (a&0xFF000000) >> 24;
+	return res;
+}
+uint64_t swap64(uint64_t a)
+{
+	uint64_t res;
+	res = (a&0x00000000000000FF) << 56;
+	res += (a&0x000000000000FF00) << 40;
+	res += (a&0x0000000000FF0000) << 24;
+	res += (a&0x00000000FF000000) << 8;
+	res += (a&0x000000FF00000000) >> 8;
+	res += (a&0x0000FF0000000000) >> 24;
+	res += (a&0x00FF000000000000) >> 40;
+	res += (a&0xFF00000000000000) >> 56;
+	return res;
+}
+uint16_t rgb888to565(uint32_t a)
+{
+	uint32_t res;
+	res = (a&0x000000F8) >> 3;
+	res += (a&0x0000FC00) >> 5;
+	res += (a&0x00F80000) >> 8;
+	return res;
+}
+uint32_t perimeter(uint32_t a,uint32_t b)
+{
+	return 2*a+2*b;
+}
+uint32_t area(uint32_t a,uint32_t b)
+{
+	return a*b;
+}
+uint32_t circle_perimeter(uint32_t a)
+{
+	return 3.14*2*a;
+}
+uint32_t circle_area(uint32_t a)
+{
+	return 3.14*a*a;
+}
+uint64_t factorial(uint16_t a)
+{
+	uint64_t res=1,i;
+	for(i=2;i<=a;i++) res *= i;
+	return res;
+}
+int print_divided_numbers(uint16_t a)
+{
+	uint16_t i;
+	for(i=a;i<500;i+=a) printf("%u   ", i);
```

Comment:
500 should have been included in this for loop:
`for(i=a;i<=500;i+=a)`

Please, use more space in such expressions in the future, e.g.
`for (i = a; i < 500; i += a)
{ 
  printf("%u   ", i);
}`